core/batcher.go

```diff
@@ -0,0 +1,348 @@
+package core
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/anoideaopen/foundation/core/cachestub"
+	"github.com/anoideaopen/foundation/core/telemetry"
+	"github.com/anoideaopen/foundation/core/types"
+	"github.com/anoideaopen/foundation/hlfcreator"
+	"github.com/anoideaopen/foundation/internal/config"
+	"github.com/anoideaopen/foundation/proto"
+	pb "github.com/golang/protobuf/proto" //nolint:staticcheck
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	BatcherBatchExecuteEvent                    = "batcherBatchExecute"
+	TxBatcherRequestType     BatcherRequestType = "tx"
+)
+
+type (
+	BatcherBatchExecuteRequestDTO struct {
+		Requests []BatcherRequest `json:"requests"`
+	}
```

Comment:
The word "Batcher" is used twice in the name, creating redundancy. Typically, one mention is enough to indicate that it relates to batch operations. The term "DTO" (Data Transfer Object) is also redundant in the Go context, where structs are assumed to be for data transfer.

```
type BatchExecuteRequest struct {
	Requests []BatchRequest `json:"requests"`
}
```

Please follow Go naming conventions for better readability. 

https://google.github.io/styleguide/go/

Reply:
Okay, I absolutely agree about DTO in naming variable, but I want to discuss about the suggested name 'BatchExecute Request' of that struct:

The "Batcher" service is service which will have group of methods.

'Batcher' prefix is the name of service and 'batchExecute' is the name of exists method check batcher will "overloading" (like java style).

Can I call method 'batcherExecute' and struct 'BatcherExecuteRequest'?

Will we see that the method 'batcherExecute' is not the same like 'batchExecute'

Reply:
we already have method name 'batchExecute' for robot and we can't call the current struct 'BatchExecuteRequest'

Reply:
Go does not support method overloading like Java, so using batcherExecute is not idiomatic. I recommend renaming the method to ExecuteBatch and the struct to ExecuteBatchRequest. If naming conflicts arise, consider moving the logic to a separate package.

https://go.dev/doc/effective_go#names
https://go.dev/wiki/CodeReviewComments#package-comments

Reply:
I have been used word "overloading" to give you information that is the two method with the same business functionality but with different method arguments. I clearly undestood that go and java it's two different lang, but "overloading" usually short word to explain the idea.

That is the reason why i put this word in quotes.

Reply:
I like the names (method ExecuteBatch, struct ExecuteBatchRequest) from your suggestion i will rename the method and structure asap

Reply:
@muzykantov I did it, thank you